app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -136,7 +136,18 @@ def available_releases
     def product_content
       content_access_mode_all = ::Foreman::Cast.to_bool(params[:content_access_mode_all])
       content_access_mode_env = ::Foreman::Cast.to_bool(params[:content_access_mode_env])
-      content = @activation_key.available_content(content_access_mode_all, content_access_mode_env)
+
+      pcf = ProductContentFinder.new(
+          :match_subscription => !content_access_mode_all,
+          :match_environment => content_access_mode_env,
+          :consumable => @activation_key
+      )
+
+      content = pcf.product_content
+
+      overrides = @activation_key.content_overrides
+      content.map! { |pc| ProductContentPresenter.new(pc, overrides) }
```

Comment:
I like the refactoring but I think you could DRY it up further by adding overrides logic to the Finder and make him return ProductContentPresenter... That would mean we can use the same logic for both AK and CH